File: public/js/game/commands/logic/moveCommand.js
function MoveCommand(sobject){
    Command.call(this, MoveCommand.ID);
    this.toData = function(){
        return {id: sobject.id, tx: sobject.targetPoint.x, ty: sobject.targetPoint.y}
    }
}
MoveCommand.ID = 'move'
Command.add(MoveCommand.ID, MoveCommand)

function MoveApplyCommand(){
    Command.call(this, MoveApplyCommand.ID);
    this.data = true
    this.execute = function(){
        var sobject = Engine.findSObject(this.data.id)
        if(sobject){
            if(sobject.targetPoint){
                sobject.targetPoint.x = this.data.tx;
                sobject.targetPoint.y = this.data.ty;
            }else
                sobject.targetPoint = new PIXI.Point(this.data.tx, this.data.ty)
        } else
            throw "moved sobject id=" + this.data.id + " not found"
    }
}
MoveApplyCommand.ID = 'move-apply'
Command.add(MoveApplyCommand.ID, MoveApplyCommand)

File: src/main/java/com/hellsepontus/commands/logic/MoveCommand.java
package com.hellsepontus.commands.logic;

import com.hellsepontus.commands.CastCommand;
import com.hellsepontus.commands.Command;

import java.util.Map;

public class MoveCommand extends CastCommand{
    
    public static final String ID = "move";
    
    public MoveCommand() {
        super(ID);
    }

    @Override
    public void execute() {
        Command responseCmd = Command.createCommand(MoveApplyCommand.ID, this);
        send(responseCmd);
    }

    public static class MoveApplyCommand extends CastCommand{

        public static final String ID = "move-apply";

        public MoveApplyCommand() {
            super(ID);
        }

        @Override
        public Map<String, Object> toData() {
            return data;
        }
    }
}
